src/main/java/course_project/battleship/Main.java

```diff
@@ -0,0 +1,11 @@
+package course_project.battleship;
+
+
+import course_project.battleship.game.Game;
+
+public class Main {
+
+    public static void main(String[] args) {
+        new Game().run();
```

Comment:
Input is not quite simple... Automatic ship placement is nice!

Reply:
Opt: Your utils folder is oversized, some classes should be moved to other places

Reply:
Quite good! You show good knowledge of Java Core and Java libraries, code is mostly clean and easy to read. Abstractions are chosen correctly, also good architecture. I didn't see any bugs in the game, but it would look better if you refactor the interface. 

Approved!